keps/693-multikueue/README.md

```diff
@@ -0,0 +1,234 @@
+is running inside the management clusters. However, as there is no job controller,
+no pods will be created in the management cluster. No controller will also overwrite
+the status that will be copied by the MultiKueue controller. 
+
+If the job, for whatever reason, is suspended or deleted in the management cluster,
+it will be deleted from the worker cluster. Deletion/suspension of the job 
+only in worker cluster will trigger the global job requeuing.
+Once the job finishes in the worker cluster, the job will also 
+finish in the management cluster.
+
+### User Stories (Optional)
+
+#### Story 1
+As a Kueue user I have clusters on different cloud providers and on-prem. 
+I would like to run computation-heavy jobs across all of them, wherever 
+I have free resources.
+ 
+#### Story 2
+As a Kueue user I have clusters in multiple regions of the same cloud 
+provider. I would like to run workloads that require the newest GPUs,
+whose on-demand availability is very volatile. The GPUs are available
+at random times at random regions. I want to use ProvisioningRequest
+to try to catch them.
+
+### Risks and Mitigations
+* Disabling the Job controller for all (or selected objects) may be problematic
+on environments where access to the master configuration is limited (like GKE).
+We are in talks with SIG-Apps to establish an acceptable way of using a non
+default controller (or none at all)
+
+## Design Details
+MultiKueue will be enabled on a cluster queue using the admission check fields.
+Just like ProvisioningRequest, MultiKueue will have its own configuration, 
+MultiKueueConfig with the following definition. To allow reusing the same clusters
+across many Kueues, additional object, MultiKueueWorkerCluster, is added.
+
+```go
+type MultiKueueConfig struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec MultiKueueConfigSpec `json:"spec,omitempty"`
+}
+
+type MultiKueueConfigSpec struct {
+     // List of MultiKueueWorkerClusters names where the 
+     // workloads from the ClusterQueue should be distributed.
+     Clusters []string `json:"clusters,omitempty"`
+}
+
+type MultiKueueCluster struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec MultiKueueClusterSpec `json:"spec,omitempty"`
+    Status MultiKueueClusterStatus `json:"status,omitempty"`
+}
+
+const (
+    // Location is the path on the disk.
+    PathLocationType LocationType = "Path"
+    
+    // Location is the name of the secret inside kueue-system namespace.
+    SecretLocationType LocationType = "Secret"
+)
+
+type MultiKueueuClusterSpec {
+    // Information how to connect to the cluster.
+    KubeConfig KubeConfig `json:"kubeConfig"`
+}
+
+type KubeConfig struct {
+    // Name of the custer inside the given KubeConfig.
+    Name string `json:"configName"`
+    
+    // Location of the KubeConfig.
+    Location string `json:"location"`
```

Comment:
In case of secretes this location should be specified as Namespace + Name + Key.

Reply:
Namespace should be fixed at kueue's namespace, otherwise Kueue would get permission to read all secrets which many users may not be comfortable with.

Reply:
We could restrict the namespace value via admission webhook or something, but since the object (etcd) has nothing to do with the namespace in which the kueue manager is running, maybe we should keep a way of specifying it.

Reply:
Can you add a note that the namespace will be fixed to the namespace where kueue is running?

Reply:
It is documented in the location type constants.